naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2474,6 +2480,29 @@ impl Parser {
 
         let mut lexer = Lexer::new(source);
         let mut tu = ast::TranslationUnit::default();
+
+        // Parse directives.
+        #[allow(clippy::never_loop, unreachable_code)]
+        while let Ok((ident, span)) = lexer.peek_ident_with_span() {
+            if let Some((kind, name)) = DirectiveKind::from_ident(ident) {
+                self.push_rule_span(Rule::Directive, &mut lexer);
+                let _ = lexer.next_ident_with_span().unwrap();
+                match kind {
+                    DirectiveKind::Unimplemented(unimplemented) => {
+                        return Err(Error::DirectiveNotYetImplemented {
+                            kind: name,
+                            span,
+                            tracking_issue_num: unimplemented.tracking_issue_num(),
+                        })
+                    }
+                }
+                self.pop_rule_span(&lexer);
+            } else {
+                break;
+            }
+            // TODO: add directives to set of expected things in errors later
```

Comment:
note: I couldn't think of a reasonable, but I think it's fine to land this MVP without it. Filed https://github.com/gfx-rs/wgpu/issues/6421 for follow-up.